libweb3jsonrpc/Debug.cpp

```diff
@@ -120,19 +142,49 @@ Json::Value Debug::debug_traceBlockByNumber(int _blockNumber, Json::Value const&
 	return ret;
 }
 
-Json::Value Debug::debug_storageAt(string const& _blockHashOrNumber, int _txIndex, string const& _address)
+Json::Value Debug::debug_storageAt(string const& _blockHashOrNumber, int _txIndex, string const& _address, string const& _begin, string const& _end, int _maxResults)
 {
 	Json::Value ret(Json::objectValue);
+	ret["complete"] = true;
+	ret["storage"] = Json::Value(Json::objectValue);
 
 	if (_txIndex < 0)
 		throw jsonrpc::JsonRpcException("Negative index");
-	Block block = m_eth.block(blockHash(_blockHashOrNumber));
+	if (_maxResults <= 0)
+		throw jsonrpc::JsonRpcException("Nonpositive maxResults");
+
+	u256 const begin(fromHexJsonRpc(_begin));
+	u256 const end(fromHexJsonRpc(_end));
+	if (begin > end)
+		throw jsonrpc::JsonRpcException("Begin is greater than end");
+
+	try
+	{
+		Block block = m_eth.block(blockHash(_blockHashOrNumber));
+
+		unsigned const i = ((unsigned)_txIndex < block.pending().size()) ? (unsigned)_txIndex : block.pending().size();
+		State state = block.fromPending(i);
+
+		map<u256, u256> const storage(state.storage(Address(_address)));
 
-	unsigned i = ((unsigned)_txIndex < block.pending().size()) ? (unsigned)_txIndex : block.pending().size();
-	State state = block.fromPending(i);
+		auto itBegin = storage.lower_bound(begin);
+		auto itEnd = storage.upper_bound(end);
```

Comment:
If end is greater than the greatest element of the map, itEnd points to storage.end()

Otherwise itEnd points to the first element greater than end (end should be inclusive)
I'll add the comment to clarify this